### ISSUE TITLE
Custom view rendering issue while converting view from Xamarin Forms to native Xamarin.Android

### ISSUE THREAD

**Issue by Swathi-Gopal** [xamarin/Xamarin.Forms#1967]
Bug report best practices: https://github.com/xamarin/Xamarin.Forms/wiki/Submitting-Issues

### Description

Adding a custom view to the UI doesn’t occupy the entire space (loaded a grid containing 3 labels Subject, Notes, Location within a frame). A frame is set with Padding property and Grid is loaded into Frame such that the grid inside the frame doesn’t occupy the space accordingly.

We have set Vertical and Horizontal Layout options to Grid. We have converted the custom view from xamarin forms to native referring the below conversion.

http://michaelridland.com/xamarin/creating-native-view-xamarin-forms-viewpage/ 

We suspect that this is a framework issue with the Grid and positioning of the elements within a Grid. Since we tried with the same layouts directly in xamarin forms and Grid occupied the entire UI in that.UI is rendered as expected Kindly find the sample along with the screenshot for this 

**Sample:** 

[CustomRenderer.zip](https://github.com/xamarin/Xamarin.Forms/files/1750631/CustomRenderer.zip)

**Screenshot :** 
![customcontrol_forms](https://user-images.githubusercontent.com/36147200/36579997-3ec78d2e-188c-11e8-9678-6c8477ee6cc7.jpg)

### Steps to Reproduce

Deploy the attached sample, in the view, a grid will be loaded in the frame according to the labels added to it. 

### Expected Behavior

Grid with labels to occupy entire Frame.

### Actual Behavior

Grid occupies the space in the Frame according to the width and height of the labels. 

**Only when width request or height request is set to grid or each of the children inside the grid, expected behaviour is obtained.**

### Basic Information

- Version with issue: Xamarin 2.4.0
- Platform Target Frameworks: 
  - Android:  7.1 

The issue is reproduced only in **Xamarin.Forms Android** platform.

### Sample 

[CustomControl.zip](https://github.com/xamarin/Xamarin.Forms/files/1750628/CustomControl.zip)

### Screenshot

![customcontrol_formstonativeconversion](https://user-images.githubusercontent.com/36147200/36580001-40f458fc-188c-11e8-8c21-6c1af7e17364.jpg)

Could you please check with the above details and provide us with a possible solution for this.

**Comment by samhouts** [comment]
The reproduction doesn't render anything on the screen for me except for the `GreenYellow` background, so I can't verify if this is the problem or not, BUT...

The Xamarin.Forms layout code uses the bounds of the Parent to determine the maximum space the View can occupy. When Views are added to Pages in Forms, the Parent is automatically set. Here, you are circumventing that by creating a `CustomView` inside your `CustomLayout` but never setting a `Parent` on it. I believe that if you set the `CustomView` or `frame's `Parent` to `this` (your `CustomLayout` reference), then the layout should be able to find the bounds of the outer `Page`. 


Please give this a try and see if that gets you closer to your desired result. Thanks!

**Comment by Swathi-Gopal** [comment]
Hi @samhouts ,

I have again attached the issue reproducing sample for your reference.

Android version – Android 7.1

Minimum Android version – API level 15

Target Android version – API 25

Xamarin Forms version – 2.4.0.280

Tested Devices - Vivo Y66, Moto E. 

**Sample:**  
[Sample.zip](https://github.com/xamarin/Xamarin.Forms/files/1766951/Sample.zip)


If I set Width and Height(bounds) to the inner child of the custom view, view is rendered as expected. Is this the only possible solution for this?

**Comment by PureWeen** [comment]
@Swathi-Gopal  it's a little hard to sift though that last attached sample. If you could get the initial minimal one to exhibit the issue then we could probably give you a more concise answer.

Also just to clarify. It sounds like what @samhouts  suggested you do worked. Is that correct? now you're just curious if there's a "better" way

**Comment by jassmith** [comment]
Okay I looked over the code and there are a number of things that are wrong here. The largest issue is CustomView does not seem to Parent its children at all, this breaks so many things. The easiest way to resolve this is to derive your control for Layout<T>. Once the view is parented we can start seeing how things are working.

The other issue I see is you are setting FillAndExpand on a grid in a frame, this does nothing.

**Comment by pauldipietro** [comment]
Closing as we've had no further responses for over a month. Please reopen with more information per Jason's last comment if the issue still occurs on 3.0/3.1-pre.

**Comment by Swathi-Gopal** [comment]
@pauldipietro ,

As mentioned in the query, actual issue is while converting the custom view from xamarin forms to native android. That is, in our case we are rendering the view alone (virtually) in Xamarin forms through mapping and layouting for the custom view and parenting its children are done in native Xamarin.Android.So when it comes to xamarin forms, parenting its children in source level is not possible ,only by setting height and width to custom view in sample level fixed the issue. Thus we have attached the sample for the mentioned issue in our earlier updates.
 
**Note:** We have checked with the latest **Xamarin Forms version 3.0** and issue exists in this version also.
